src/main/java/org/terasology/web/resources/systemStatus/SystemMetadata.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.web.resources.systemStatus;
+
+import oshi.SystemInfo;
+
+public final class SystemMetadata {
+
+    private double cpuUsage;
+    private double memoryUsagePercentage;
+    private long memoryUsed;
+    private long memoryTotal;
+    private long memoryAvailable;
+    private long systemUptime;
+
+    public static SystemMetadata build() {
+        SystemMetadata systemMetadata = new SystemMetadata();
+        systemMetadata.getStatus();
+        return systemMetadata;
+    }
+
+    private void getStatus() {
+        SystemInfo systemInfo = new SystemInfo();
```

Comment:
Oh and to go on with the micro-optimizations, this might be stored as an instance field, i.e. `private SystemInfo systemInfo = new SystemInfo();` and only initialized once together with this class (in the declaration, i.e. no other `new`s besides the one in the snippet I just wrote), **but** the implementation of `SystemInfo` may either
1. read the data from the system when the instance is constructed, store it in its internal fields and then always return the same data when the getters are called, or
2. obtain the data every time the getters are called
Obviously, my suggestion would only make sense if it works as in option 2. Having a look at oshi's docs/source and/or making some empirical observation whether we always get the same data could be helpful to understand if it works like 1 or 2 and thus if it makes sense to apply this change.

However if it's too much work it may not be worth it. These are mostly random thoughts and is by no means "requested changes"

And finally - in the local variable or instance field (`systemInfo` but name should we changed as well if the type is changed) you may store the result of `getHardware()` since it's its only member we need.
My reasoning is that since oshi is an external library we don't really know what `getHardware()` actually does - it may make expensive calculations (I don't think it's the case but who knows) so if we can reduce the amount of calls we do to external code why not? (of course there may be counterarguments, happy to hear them if anyone has any)